pkg/operator/ceph/cluster/crash/crash_test.go

```diff
@@ -19,11 +19,55 @@ package crash
 import (
 	"testing"
 
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned/fake"
+	"github.com/rook/rook/pkg/client/clientset/versioned/scheme"
+	"github.com/rook/rook/pkg/clusterd"
+	cephver "github.com/rook/rook/pkg/operator/ceph/version"
+	"github.com/rook/rook/pkg/operator/test"
 	"github.com/stretchr/testify/assert"
+	"github.com/tevino/abool"
+	v1 "k8s.io/api/batch/v1"
+	"k8s.io/api/batch/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
 func TestGenerateCrashEnvVar(t *testing.T) {
 	env := generateCrashEnvVar()
 	assert.Equal(t, "CEPH_ARGS", env.Name)
 	assert.Equal(t, "-m $(ROOK_CEPH_MON_HOST) -k /etc/ceph/crash-collector-keyring-store/keyring", env.Value)
 }
+
+func TestCreateOrUpdateCephCron(t *testing.T) {
+	cephCluster := cephv1.CephCluster{ObjectMeta: metav1.ObjectMeta{Namespace: "rook-ceph"}}
+	cephVersion := &cephver.CephVersion{Major: 16, Minor: 2, Extra: 0}
+
+	// Objects to track in the fake client.
+	context := &clusterd.Context{
+		Clientset:                  test.New(t, 1),
+		RookClientset:              rookclient.NewSimpleClientset(),
+		RequestCancelOrchestration: abool.New(),
+	}
+
+	s := scheme.Scheme
+	err := v1beta1.AddToScheme(s)
+	if err != nil {
+		assert.Fail(t, "failed to build scheme")
+	}
+
+	err = v1.AddToScheme(s)
+	if err != nil {
+		assert.Fail(t, "failed to build scheme")
+	}
+
+	r := &ReconcileNode{
+		scheme:  s,
+		client:  fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects().Build(),
+		context: context,
+	}
+
+	controllerutil, err := r.createOrUpdateCephCron(cephCluster, cephVersion, true)
```

Comment:
suggestion: it should test both k8s version less than 1.21.0 and greater. Also assert that correct cronjob versions got created.

Reply:
done.

Reply:
IMO, the test flow should be like this:
1. call `createOrUpdateCephCron` with ` v1Beta1`. Then assert that only `v1Beta1` is present  and not `v1`. 
2. call `createOrUpdateCephCron` again with `v1`. Then assert that only `v1` is created and the older `v1Beta1` versions are deleted. 

Rather than checking `cntrlutil.OperationResult("unchanged")` or `cntrlutil.OperationResult("created")),` we should use the fake client to check if the `v1` or `v1Beta1` object got created or not.